app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java

```diff
@@ -0,0 +1,223 @@
+            return;
+        }
+        currentLocation = location;
+        displayLocation();
+    }
+
+    private void displayLocation() {
+        if (currentLocation != null) {
+            LatLng latLng = new LatLng(currentLocation.getLatitude(), currentLocation.getLongitude());
+            CameraUpdate cameraUpdate = CameraUpdateFactory.newLatLngZoom(latLng, 14);
+            map.animateCamera(cameraUpdate);
+        }
+    }
+
+
+    private List<Report> queryReports() {
+        List<Report> reports = new ArrayList<>();
+        ParseQuery<Report> query = ParseQuery.getQuery(Report.class);
+        query.include(Report.TYPE_OF_CRIME_KEY);
+        query.setLimit(50);
+        query.addDescendingOrder("createdAt");
+        query.findInBackground((reportList, e) -> {
+            if (e != null) {
+                Log.e(TAG, "Issue with getting posts", e);
+                return;
+            }
+            reports.addAll(reportList);
+            showReports(reports);
+        });
+        return reports;
+    }
+
+    private void showReports(List<Report> reports) {
+        for (Report report : reports) {
+            showMarker(report.getTypeOfCrime().getTag(),
+                    report.getLocation().getLatitude(),
+                    report.getLocation().getLongitude(),
+                    report.getTypeOfCrime().getLevelOfRisk());
+        }
+    }
+
+    private void showMarker(String title, double latitude, double longitude, int levelOfRisk) {
+        LatLng reportLatLng = new LatLng(latitude, longitude);
+        Marker marker;
+        MarkerOptions markerOptions = new MarkerOptions()
+                .position(reportLatLng)
+                .title(title).icon(getNewIcon(levelOfRisk));
+        marker = map.addMarker(markerOptions);
+        dropPinEffect(marker);
+    }
+
+    private void dropPinEffect(Marker marker) {
+        // Handler allows us to repeat a code block after a specified delay
+        final android.os.Handler handler = new android.os.Handler();
+        final long start = SystemClock.uptimeMillis();
+        final long duration = 1500;
+        // Use the bounce interpolator
+        final android.view.animation.Interpolator interpolator =
+                new BounceInterpolator();
+        // Animate marker with a bounce updating its position every 15ms
+        handler.post(new Runnable() {
+            @Override
+            public void run() {
+                long elapsed = SystemClock.uptimeMillis() - start;
+                // Calculate t for bounce based on elapsed time
+                float t = Math.max(
+                        1 - interpolator.getInterpolation((float) elapsed
+                                / duration), 0);
+                // Set the anchor
+                marker.setAnchor(0.5f, 1.0f + 14 * t);
+
+                if (t > 0.0) {
+                    // Post this event again 15ms from now.
+                    handler.postDelayed(this, 15);
```

Comment:
Might be worth considering a PinEffect class to contain the animation data.

I'd also pull the duration values (1500, 15) out into static final variables so you can see them at the top and allow for easy modification to behavior later.